Core/HLE/proAdhoc.cpp

```diff
@@ -2211,7 +2211,7 @@ int initNetwork(SceNetAdhocctlAdhocId *adhoc_id){
 			u64 now = (u64)(time_now_d() * 1000000.0);
 			if (coreState == CORE_POWERDOWN) 
 				return iResult;
-			if (now - startTime > (getSockError((int)metasocket) == EINPROGRESS ? adhocDefaultTimeout*2LL: adhocDefaultTimeout))
+			if (now - startTime > (u64)(getSockError((int)metasocket) == EINPROGRESS ? adhocDefaultTimeout*2LL: adhocDefaultTimeout))
```

Comment:
I was going to fix the warning issue on this line on the PR i'm currently working, but i'm using `static_cast<s64>(now - startTime)` because we were having an issue where time can goes backward before on ScaledTime (resulting negative value to be a very large positive value).
So it might be better to use signed type for delta value, thus the comparison won't get messed up in such case.

Reply:
That's a good point. Go ahead fix it your proposed way!

Reply:
I mean, we need to fix whatever is trying to read the time on another thread, because that's just unsafe.  Time should be treated as an unsigned value everywhere.

-[Unknown]